### ISSUE TITLE
Too strict equality check / too vague message for undefined properties

### ISSUE THREAD

**Issue by dzonatan** [CodeSequence/jasmine-marbles#84]
Before `v0.8.1` this was passing ✅ :
```ts
const actions = of({ propA: 1, propB: undefined });

// take attention that `propB` is not explcitly defined in the expected object
expect(actions).toBeObservable(cold('(a|)', { a: { propA: 1 } }));
```

Starting with `v0.8.1` it's failing ❌ with an assertion message:

```
Expected: (a|),
Received: (?|),

Expected:
[{"frame":0,"notification":{"kind":"N","value":{"propA":1},"hasValue":true}},{"frame":0,"notification":{"kind":"C","hasValue":false}}]

Received:
[{"frame":0,"notification":{"kind":"N","value":{"propA":1},"hasValue":true}},{"frame":0,"notification":{"kind":"C","hasValue":false}}]

```
which, as you can see, doesn't tell exactly why it's failing. The `expected` and `received` messages are identical. 
It took me a while to find out what's wrong...

#65 seems to be the cause of this equality check change - `lodash.isEqual` strictly checks the defined properties as well.

Either this needs to be reverted back to ignore the `undefined` properties or the error message needs to be improved so it's clear how the objects differ.

I would go with the first option just to be in line with how the jest itself does a deep check:
```ts
expect({ propA: 1, propB: undefined }).toEqual({ propA: 1 }); // passes ✅
```

EDIT: Seems like  #76 is going to fix this.

**Comment by brandonroberts** [comment]
Closed via #86

**Comment by brandonroberts** [comment]
Fix released in `0.8.4`
